engine/src/core/utils/file_manager.h

```diff
@@ -18,9 +19,15 @@ namespace Bald::Utils {
         /**
          * ENUM which determines size of file and therefor methods, which will be used to read it
          */
-        enum class Size : char {
+        enum class Size : uint8_t {
             SMALL_FILE, BIG_FILE
         };
+
+        enum class Error : uint8_t {
```

Comment:
Let's discuss about the error type. I think having a global type would be interesting as well (we could write a constructor overload for this error type and be able to return from function Error::SomeError instead of unexpected(FileManager::Error::CantOpenFile))